app/src/main/java/com/example/dungeontest/Settings.kt

```diff
@@ -123,22 +225,102 @@ fun SimpleOutlinedTextFieldSample() {
         )
     }
     val textField = FocusRequester()
+    val focusManager = LocalFocusManager.current
+    val textFieldWidth = (maxLength * 8).dp
     OutlinedTextField(
-
-        value = text,
+        value = tokenValue.value,
         onValueChange = {
-            if (it.length <= maxLength) text = it
-            if (it.length == 2) {
-                textField.freeFocus()
-
-            }
+            tokenValue.value = it
+//            Log.d("SimpleOutlinedTextFieldSample", "New value: $text")
         },
 
-        modifier = Modifier.focusRequester(textField),
+        modifier = Modifier.focusRequester(textField).width(textFieldWidth),
         singleLine = true,
         shape = MaterialTheme.shapes.large,
         textStyle = TextStyle(brush = brush),
-        label = { Text("Your OpenAI Key") }
+        label = { Text("Your OpenAI Key") },
+        keyboardActions = KeyboardActions(
+            onDone = {
+                focusManager.clearFocus()
+            }
+        ),
 
     )
-}
\ No newline at end of file
+}
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+fun OutlinedCardExample(id: Int, title: String, description: String, selectedCard: MutableState<Int>) {
+    val keyboardController = LocalSoftwareKeyboardController.current
+    val focusManager = LocalFocusManager.current
+
+    OutlinedCard(
+
+        modifier = Modifier,
+        border = BorderStroke(
+            width = 2.dp,
+            color = if (selectedCard.value == id) MaterialTheme.colorScheme.primary else Color.Gray
+        ),
+
+    ) {
+        Column(modifier = Modifier
+            .clickable {
+                selectedCard.value = id
+                keyboardController?.hide()
+                focusManager.clearFocus()
+            }
+            .defaultMinSize(200.dp, 140.dp)
+        ){
```

Comment:
why are we nesting a single Row inside a Column? Why not just give the current complex modifier (lines 266-272) to the Row composable call (currently line 274) instead?

Reply:
To arrange the two child columns horizontally

Reply:
But wouldn't that be achieved as well by having a Row with 2 Columns inside it, without another Column outside/wrapping-around the Row?

Reply:
@BareBeaverBat Addressed. Remove the Row from inside the single Column. Moved the outlined modifiers appropriately as well

Reply:
@BareBeaverBat Nope. I retrack. I just remembered why I added the column. Having the outer parent column is needed ensure the ripple effect's corners are curved. 
Restored the parent column but moved all styling modifiers to the child row composable.